### PR TITLE
use 0 dim tensor

### PR DESCRIPTION
**Pull request by liweiwp**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1118/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1118/

**Comment by liweiwp** [comment]
> Could you explain why the changes are necessary or better than the existing code? IMO explicit is better than implicit.

@terrytangyuan @AnirudhDagar 
For the dim/size,  in MXNet version, the variable a is a scalar, but the Pytorch version is a one element vector, and the TensorFlow version is a one element matrix, which confused beginners like me, so I think the three version should consistent
For the dtype,  I think is not need for this demo, for compact code make the us focus on the key points, also the three version should consistent
